modules/sugar/src/main/ts/ephox/sugar/api/dom/DomFuture.ts

```diff
@@ -1,15 +1,11 @@
-import { clearTimeout, setTimeout } from '@ephox/dom-globals';
+import { clearTimeout, Event, setTimeout } from '@ephox/dom-globals';
 import { Future, LazyValue, Result } from '@ephox/katamari';
 import * as DomEvent from '../events/DomEvent';
 import { EventArgs } from '../events/Types';
 import Element from '../node/Element';
 
-type WorkDone = (res: Result<EventArgs, string>) => void;
-type Worker = (callback: WorkDone) => void;
-type TaskConstructor<T> = (worker: Worker) => T;
-
-const w = <T> (fType: TaskConstructor<T>, element: Element, eventType: string, timeout: number) => fType((callback) => {
```

Comment:
Yeah, this function was interesting. You can only give it a correct type if you have higher-kinded types.